backend/src/main/scala/com/softwaremill/adopttapir/starter/api/StarterRequestValidator.scala

```diff
@@ -125,6 +133,18 @@ sealed trait FormValidator {
     }
   }
 
+  private def validateEffectImplementationWithMetrics(
```

Comment:
What you think about returning only the validated part? 

```
  private def validateMetricsWithEffectAndImplementation(
      effect: EffectRequest,
      serverImplementation: ServerImplementationRequest,
      addMetrics: Boolean
  ): ValidatedNec[RequestValidation, Boolean] = {
    (effect, serverImplementation, addMetrics) match {
      case t @ (_, Akka | Http4s | ZIOHttp, _) => t._3.validNec
      case t @ (_, Netty, false)               => t._3.validNec
      case (_, Netty, true) => RequestValidation.MetricsNotSupportedForNettyServerImplementation(effect, serverImplementation).invalidNec
    }
  }
```

Then you can reuse result. 
```
  def validate(r: StarterRequest): Either[IncorrectInput, StarterDetails] =
    (
      validateSemanticVersioning(r.tapirVersion),
      validateProjectName(r.projectName),
      validateGroupId(r.groupId),
      validateEffectWithImplementation(r.effect, r.implementation),
      validateMetricsWithEffectAndImplementation(r.effect, r.implementation, r.addMetrics),
      validateEffectWithJson(r.effect, r.json)
    ).mapN { case (tapirVersion, projectName, groupId, (effect, serverImplementation), addMetrics, json) =>
      StarterDetails(
        projectName,
        groupId,
        effect.toModel,
        serverImplementation.toModel,
        tapirVersion,
        r.addDocumentation,
        addMetrics,
        json.toModel
      )
    }.leftMap(errors => IncorrectInput(errors.toNonEmptyList.map(_.errMessage).toList.mkString(System.lineSeparator())))
      .toEither
```

Reply:
Good point! Was wondering how to achieve it ;)